sdk/component/store/remote/postgresql/store_test.go

```diff
@@ -0,0 +1,313 @@
+package postgresql_test
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/ory/dockertest/v3"
+	"github.com/ory/dockertest/v3/docker"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	"github.com/smithy-security/smithy/sdk/component/store"
+	"github.com/smithy-security/smithy/sdk/component/store/remote/postgresql"
+	"github.com/smithy-security/smithy/sdk/component/uuid"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+	ocsf "github.com/smithy-security/smithy/sdk/gen/ocsf_schema/v1"
+)
+
+type (
+	Storer interface {
+		component.Storer
+		Ping(ctx context.Context) error
+	}
+
+	ManagerTestSuite struct {
+		suite.Suite
+
+		pool          *dockertest.Pool
+		psqlResource  *dockertest.Resource
+		atlasResource *dockertest.Resource
+		dockerNetwork *docker.Network
+
+		manager Storer
+	}
+)
+
+func TestManagerTestSuite(t *testing.T) {
+	suite.Run(t, new(ManagerTestSuite))
+}
+
+func (suite *ManagerTestSuite) SetupSuite() {
+	const (
+		user     = "smithy"
+		password = "smithy1234"
+		dbName   = "smithy-findings"
+	)
+
+	var (
+		err         error
+		ctx, cancel = context.WithTimeout(context.Background(), time.Minute)
+		clock       = clockwork.NewFakeClockAt(
+			time.Date(2024, 11, 1, 0, 0, 0, 0, time.UTC),
+		)
+	)
+	defer cancel()
+
+	suite.pool, err = dockertest.NewPool("")
+	require.NoError(suite.T(), err)
+
+	suite.dockerNetwork, err = suite.pool.Client.CreateNetwork(docker.CreateNetworkOptions{
+		Name:   "test-network",
+		Driver: "bridge",
+	})
+	require.NoError(suite.T(), err)
+
+	suite.psqlResource, err = suite.pool.RunWithOptions(&dockertest.RunOptions{
+		Name:       "postgres",
+		Platform:   "linux/amd64",
+		Repository: "postgres",
+		Tag:        "15",
+		Env: []string{
+			fmt.Sprintf("POSTGRES_USER=%s", user),
+			fmt.Sprintf("POSTGRES_PASSWORD=%s", password),
+			fmt.Sprintf("POSTGRES_DB=%s", dbName),
+		},
+		NetworkID: suite.dockerNetwork.ID,
+	}, func(config *docker.HostConfig) {})
+
+	var (
+		psqlDockerPort = suite.psqlResource.GetPort("5432/tcp")
+		psqlDSN        = fmt.Sprintf(
+			"postgresql://%s:%s@localhost:%s/%s?sslmode=disable&connect_timeout=10",
+			user,
+			password,
+			psqlDockerPort,
+			dbName,
+		)
+	)
+
+	require.NoError(suite.T(), suite.pool.Retry(func() error {
+		pingCtx, pingCancel := context.WithTimeout(ctx, time.Second)
+		defer pingCancel()
+		suite.manager, err = postgresql.NewManager(
+			ctx,
+			postgresql.ManagerWithClock(clock),
+			postgresql.ManagerWithConnDSN(psqlDSN),
+		)
+		if err != nil {
+			return err
+		}
+		return suite.manager.Ping(pingCtx)
+	}))
+
+	p, err := filepath.Abs(".")
```

Comment:
I didn't add code to run migrations really - I relied on the ready atlas container

Let's look into this once we work on smithyctl because we need a centralised way to look into it as you mentioned :)